File: Chapter 03/C#/Section_3_3_5_DIP/Before/Midfielder.cs
using System;

namespace Chapter03.DIP.Before;

/// <summary>
/// A Low-Level Module.
/// 
/// ARCHITECTURE NOTE: Similar to the Forward, this concrete class is 
/// directly wired into the Coach. There is no abstraction, meaning the 
/// system is rigid and difficult to scale.
/// </summary>
public class Midfielder
{
    public void ControlMidfield() => Console.WriteLine("  [Action] Midfielder is controlling the game.");
}

File: Chapter 03/JS/section_3_3_5_dip/before/midfielder.js
/**
 * A Low-Level Module.
 * * ARCHITECTURE NOTE: Similar to the Forward, this concrete class is 
 * directly coupled to the Coach. There is no "seam" or abstraction 
 * between the two.
 */
class Midfielder {
    controlMidfield() {
        console.log("  [Action] Midfielder is controlling the game.");
    }
}

module.exports = Midfielder;